File: lab1/lab1_a2.m
%正弦序列
n=-20:20;
xn1=sin(pi*n/20);          
xn2=sin(pi*n/5);
subplot(2,1,1);stem(n,xn1,'.');
title('正弦序列，周期拓展两倍');
subplot(2,1,2);stem(n,xn2,'.');
title('正弦序列，周期压缩一半');

File: lab1/lab1_a2.py
import matplotlib.pyplot as plt
import math

n=[i for i in range(-20,20)]
xn1=[math.sin(3.14*i/20) for i in n];
xn2=[math.sin(3.14*i/5) for i in n];
plt.subplot(2,1,1)
plt.stem(n,xn1,linefmt='r-',markerfmt='r.',basefmt='b-')
plt.title('xn1')
plt.subplot(2,1,2)
plt.stem(n,xn2,linefmt='r-',markerfmt='r.',basefmt='b-')
plt.title('xn2')
plt.show()
